keycloak-create-test-users/src/main/java/CsvUserImport.java

```diff
@@ -0,0 +1,130 @@
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * Load users from a CSV file
+ * Expected format:
+ * First row headers
+ * Col 1 - LDAP username (org-userid)
+ * Col 2 - Firstname Lastname
+ * Col 3 - Role
+ * Col 4 - email
+ * Col 5 - last modified (ignored)
+ * Col 6 - email domain (ignored)
+ * Col 7 - org name (ignored)
+ * Col 8 - Keycloak username
+ * @author greg.perkins
+ */
+public class CsvUserImport {
+    private static Map<String,String> orgMap;
+    private static String configPath;
+    private static UserService userService;
+    private static final Logger LOG = Logger.getLogger(CsvUserImport.class.getName());
+    
+    /**
+     * Main method - runs the import
+     * @param args
+     * @throws IOException
+     * @throws URISyntaxException 
+     */
+    public static void main(String[] args) throws IOException, URISyntaxException{
+        InputStream is = CsvUserImport.class.getResourceAsStream("/idir.csv");
+        //InputStream is = CsvUserImport.class.getResourceAsStream("/test.csv");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+        String line = reader.readLine(); //ignore first line
+        line = reader.readLine(); 
+        List<User> users = new ArrayList<>();
+        while (line!=null){
+            String[] parts = line.split(",");
+            if (parts.length>=8){
+                User user = parseUserFromLine(parts, "PLR");
```

Comment:
This `"PLR"` could be defined by a variable above (`String clientId = "PLR";`) to make it a bit more clear as to what needs to be edited before running this class.